partiql-tests-data/eval/rfc/0007.ion

```diff
@@ -0,0 +1,224 @@
+            result: EvaluationSuccess,
+            output: $bag::[
+                { title: "SDE" },
+                { title: "Principal SDE" },
+            ]
+        }
+    },
+    {
+        name: "Example 6 — Value Coercion",
+        statement: '''SELECT * FROM << 1 >>
+                      UNION
+                      SELECT * FROM << 'A' >>''',
+        assert: {
+            evalMode:[EvalModeCoerce, EvalModeError],
+            result: EvaluationFail,
+        }
+    },
+
+]
+
+
+bag::[
+    {
+        name: "Example 2.3 — Union of Compatible Relations; Mismatch Column Names; Using OUTER UNION",
+        statement: '''SELECT name, title FROM hr.employees
+                      OUTER UNION ALL
+                      SELECT name, title FROM engineering.employees''',
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[
+                { name: "Bob Smith", title: "Consultant" },
+                { name: "Susan Smith", title: "Manager" },
+                { name: "Jane Smith",  title: "DEI" },
+                { title: "Manager", name: "Paul McCharmly" },
+                { title: "SDE", name: "George Parasol" },
+                { title: "Principal SDE", name: "Ringo Stone" },
+                { title: "Intern", name: "Eric Lennon" }
+            ]
+        }
+    },
+    {
+        name: "Example 3 — Union of Heterogenous Relations",
+        statement: '''SELECT * FROM hr.employees               -- T1
+                      UNION
+                      SELECT title FROM engineering.employees  -- T2''',
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[
+                { name: "Bob Smith", title: "Consultant" },
+                { name: "Susan Smith", title: "Manager" },
+                { name: "Jane Smith",  title: "DEI" },
+                { title: "Manager" },
+                { title: "SDE" },
+                { title: "Principal SDE" },
+                { title: "Intern" }
+            ]
+        }
+    },
+    {
+        name: "Example 6 — Value Coercion; Coercion of single value",
+        statement: '''SELECT * FROM << 1 >> OUTER UNION 'A' ''',
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[
+                { "_1": 1 },
+                "A",
+            ]
+        }
+    },
+    {
+        name: "Example 6 — Value Coercion",
+        statement: '''SELECT * FROM << 1 >>
+                      OUTER UNION
+                      SELECT * FROM << 'A' >>''',
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[
+                { "_1": 1 },
+                { "_1": "A" },
+            ]
+        }
+    },
+    {
```

Comment:
Could we also include the example 7 test case failure:

```
TABLE engineering.employees UNION << MISSING >>
-- ERROR!
```

And the following test case:

```
TABLE engineering.employess OUTER UNION << MISSING >>
<<
  {
    'title': 'Manager',
    'name': 'Paul McCharmly'
  },
  {
    'title': 'SDE',
    'name': 'George Parasol'
  },
  {
    'title': 'Principal SDE',
    'name': 'Ringo Stone'
  },
  {
    'title': 'Intern',
    'name': 'Eric Lennon'
  }
  MISSING
>>
-- OK!
```